File: lesson02/app/src/main/java/br/odb/nehe/lesson02/GL2JNILib.java
package br.odb.nehe.lesson02;

import android.content.res.AssetManager;

public class GL2JNILib {

    public static AssetManager assetManager;

     static {
         System.loadLibrary("NdkGlue");
     }

    /**
     * @param width the current view width
     * @param height the current view height
     */
    public static native void init(int width, int height);
    public static native void step();
    public static native void onDestroy();
    public static native void onCreate( AssetManager assetManager );
}

File: lesson02/app/src/main/java/br/odb/nehe/lesson02/GL2JNILib.kt
package br.odb.nehe.lesson02

import android.content.res.AssetManager

internal object GL2JNILib {

	/**
	 * @param width the current view width
	 * @param height the current view height
	 */
	@JvmStatic
	external fun init(width: Int, height: Int)

	@JvmStatic
	external fun step()

	external fun onDestroy()
	external fun onCreate(assetManager: AssetManager?)

	init {
		System.loadLibrary("NdkGlue")
	}
}